core/feed_protocol/types.py

```diff
@@ -0,0 +1,160 @@
+from typing import Any, Dict, Generator, List, Optional, Tuple
+
+from pydantic import BaseModel
+
+from core.model import LicensePool, Work
+from core.model.edition import Edition
+from core.model.identifier import Identifier
+
+
+class FeedEntryType(BaseModel):
+    text: Optional[str] = None
+
+    class Config:
+        extra = "allow"
+        arbitrary_types_allowed = True
+
+    def add_attributes(self, attrs: Dict[str, Any]) -> None:
+        for name, data in attrs.items():
+            setattr(self, name, data)
+
+    def children(self) -> Generator[Tuple[str, "FeedEntryType"], None, None]:
+        """Yield all FeedEntryType attributes"""
+        for name, value in self:
+            if isinstance(value, self.__class__):
+                yield name, value
+        return
+
+
+class Link(FeedEntryType):
```

Comment:
It seems like we should be basing our types here on the OPDS 2.x spec, so that our data model maps neatly onto OPDS 2.x. 

That would make this class correspond to the [readium webpub `Link` object](https://readium.org/webpub-manifest/#24-the-link-object).

Reply:
That's a lot of additional data which is not used under #1340 . Currently I wouldn't know when or how to use the additional data, we can address this change under #1340 .

Reply:
Sure. I'll resolve this one.